### ISSUE TITLE
Day planner removes input

### ISSUE THREAD

**Issue by tnljba2** [ivan-lednev/obsidian-day-planner#107]
- day planner on file mode on ios all latest sw new iphone latest obsidian
- generates daily planner fill in day plan
- day planner generates mermaid gantt all looks well
- moments later all but first entry rest is gone without a reason
![EC63319F-1CFE-46FB-B81F-FAC071FFF96F](https://user-images.githubusercontent.com/79747583/128983395-80169099-6a15-4b2e-9a6e-f6e9b7f65bd5.png)

**Comment by vslotman** [comment]
I noticed this as well on Windows and Android, but with slightly different behavior.

Windows: Sometimes when editing a task-entry the whitespace-characters are removed, resultinginaverylongstring
Android: Noticed one instance where my entire note got removed. This happened after cutting-pasting a task-entry.

On Windows these issues are resolved by disabling the Mermaid Gantt-option. Have not tried Android yet.

I have a hunch that this might be related to Obsidian noticing external changes to the file, and merging the content. I get a notification about this every time I edit a task-line with the Gantt-option enabled.

Would it be possible to only update the Gantt-chart after certain events? Just some ideas:
- When switching the note from Edit to Preview-mode
- When closing the note
- When switching focus to a different note
